src/main/java/com/example/demo/controllers/AuctionController.java

```diff
@@ -0,0 +1,28 @@
+package com.example.demo.controllers;
+
+import com.example.demo.dtos.AuctionDTO;
+import com.example.demo.entities.User;
+import com.example.demo.services.AuctionService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+public class AuctionController {
+    private final AuctionService auctionService;
+
+    @PostMapping("/auction")
+    public void createAuction(@AuthenticationPrincipal User owner,
+                                        @RequestBody AuctionDTO dto,
+                                        @RequestParam Float startPrice,
+                                        @RequestParam String title){
+        auctionService.createAuction(owner,dto,startPrice,title);
+    }
+
+    @PutMapping("/bid/{auction_id}")
+    public HttpStatusCode placeBidToAuction(@AuthenticationPrincipal User pretender,@RequestParam Float price,@PathVariable Long auction_id){
```

Comment:
Please don't return status code as a return type/value.

Status codes are controller by respective spring annotations. In your case this should be 201 or 204. By default spring returns 200, so you should set status explicitly.

Reply:
Used service method returns 404, 412 or 202 depending on inside logic. So there is no default/always-returning status. What exactly is the problem?